mindsdb/integrations/handlers/prometheus_handler/prometheus_handler.py

```diff
@@ -0,0 +1,237 @@
+            raise
+        except Exception as unknown_error:
+            logger.error(
+                f"Unknown error when connecting to Prometheus: {unknown_error}"
+            )
+            raise
+
+    def disconnect(self):
+        """
+        Closes the connection to the Prometheus host if it's currently open.
+        """
+
+        if not self.is_connected:
+            return
+
+        del self.connection
+        self.is_connected = False
+
+    def check_connection(self) -> StatusResponse:
+        """
+        Checks the status of the connection to the Prometheus host.
+
+        Returns:
+            StatusResponse: An object containing the success status and an error message if an error occurs.
+        """
+        response = StatusResponse(False)
+        connection = self.connect()
+        ok = connection.check_prometheus_connection()
+        response.success = ok
+
+        if not ok:
+            logger.error("Error connecting to Prometheus!")
+
+        return response
+
+    def native_query(self, query: Text, limit: Optional[int] = None) -> Response:
+        """
+        Executes a native PromQL query on the Prometheus host and returns the result.
+
+        Args:
+            query (str): The PromQL query to be executed.
+
+        Returns:
+            Response: A response object containing the result of the query or an error message.
+        """
+        response = Response(RESPONSE_TYPE.TABLE, False)
+        need_to_close = self.is_connected is False
+        connection = self.connect()
+
+        try:
+            result = connection.custom_query(query)
+
+            rows = []
+            for row in result:
+                rows.append(row["metric"])
+
+            df = pd.DataFrame(rows)
+            if "__name__" in df.columns:
+                df = df.drop(["__name__"], axis=1)
+            response.data_frame = df
+        except Exception as error:
+            logger.error(f'Unknown error running query: {query} on Prometheus, {error}!')
+            response = Response(
+                RESPONSE_TYPE.ERROR,
+                error_message=str(error)
+            )
+
+        if need_to_close:
+            self.disconnect()
+        elif not self.is_connected:
+            self.is_connected = False
+
+        return response
+
+    def query(self, query: ast.ASTNode) -> Response:
+        """
+        Executes a SQL query represented by an ASTNode on the Prometheus host and retrieves the data.
+        Currently, only SELECT statements with WHERE and AND are supported.
+
+        Args:
+            query (ASTNode): An ASTNode representing the SQL query to be executed.
+
+        Returns:
+            Response: The response from the `native_query` method, containing the result.
+        """
+        if not isinstance(query, Select):
+            raise ValueError("Only select queries are supported")
+
+        select_query = cast(Select, query)
+        from_table = select_query.from_table
```

Comment:
Nit: The from_table variable does not seem to be used; should we maybe use in the subsequent lines rather than calling `select_query.from_table` again?